src/bin/observable.ts

```diff
@@ -292,6 +298,12 @@ function helpArgs<T extends DescribableParseArgsConfig>(
 
   // Log automatic help.
   if ((result.values as any).help) {
+    // omit hidden flags from help
+    for (const key of Object.keys(options)) {
+      if (options[key].hidden) {
+        delete options[key];
```

Comment:
This is mutating the passed-in `options`, which might be surprising (and is generally a pattern I try to avoid). Maybe we could use a pattern like this instead?

```js
const publicOptions = Object.fromEntries(Object.entries(options).filter(([, option]) => !option.hidden));
```

And then replace references to `options` with `publicOptions` as appropriate?

Reply:
fwiw I only mutated because we are process.exit-ing a few lines later, but I agree it could be surprising. Changed!